### PR TITLE
Build: Use wildcard version for Hive and Tez

### PR DESCRIPTION
**Pull request by marton-bod**
This PR will extract the Hive and Tez versions into a variable and have the projects use these variables.
This helps: 
- remove duplication from the `versions.props` file (it has 4 different Hive entries and 2 Tez entries currently)
- remove potential for mistake of adding new Hive dependencies with the wrong version number

### CODE REVIEW

File: hive3/build.gradle [hunk 1]
```diff
@@ -16,6 +16,9 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+def hiveVersion = "3.1.2"
```
Comment (pvary) [inline hive3/build.gradle:19]: Maybe a comment here that we override the original values for hive3?

Comment (marton-bod) [reply hive3/build.gradle:21]: Sure, done

**Comment by marton-bod** [comment]
@rdblue can you please take a look as well when you have some time? Thanks so much

File: versions.props [hunk 2]
```diff
@@ -30,10 +29,7 @@ org.projectnessie:* = 0.9.2
 org.junit.vintage:junit-vintage-engine = 5.7.2
 org.junit.jupiter:* = 5.7.2
 org.mockito:mockito-core = 3.7.7
-org.apache.hive:hive-exec = 2.3.8
-org.apache.hive:hive-service = 2.3.8
 org.apache.tez:tez-dag = 0.8.4
-org.apache.tez:tez-mapreduce = 0.8.4
```
Comment (rdblue) [inline versions.props:36]: Why not do this with wildcards instead? Isn't this the same as setting `org.apache.hive:* = 2.3.8`?

**Comment by rdblue** [comment]
Are you sure this is needed? I think a couple wildcards in versions.props could accomplish the same thing?

**Comment by marton-bod** [comment]
@rdblue Thanks Ryan! I seem to remember having issues in the past with using wildcards for Hive specifically -  when using `*`, it forced the 2.3.8 version on all the transitive org.apache.hive dependencies as well, and one of them didn't have this version available. But checking it again, it seems that this issue is gone, so I've changed the PR to use the wildcards for Hive and Tez. Thanks for the suggestion!

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks for updating it, @marton-bod! I merged this.

**Pull request merged**